### PR TITLE
#145 create a bonfhir Gluestack UI package

### PR DESCRIPTION
**Pull request by nickvallee**
linked to #145 

- Introduce Gluestack UI package and FhirValue

### CODE REVIEW

File: packages/gluestack-ui/src/r5/renderer.ts [hunk 1]
```diff
@@ -0,0 +1,8 @@
+import { FhirUIRenderer } from "@bonfhir/react/r5";
+
+import { GlueStackFhirValue } from "./data-display/index";
+
+export const GluestackUIRenderer: Partial<FhirUIRenderer> = {
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  FhirValue: GlueStackFhirValue as any,
```
Comment (nickvallee) [inline packages/gluestack-ui/src/r5/renderer.ts]: I will address the type stuff in later PRs because It was requested that I submit this PR as is.

Comment (julienblin) [reply packages/gluestack-ui/src/r5/renderer.ts]: I don't understand why this is `any` here. Just remove it and you should be fine.

File: packages/gluestack-ui/src/r5/index.ts [hunk 2]
```diff
@@ -0,0 +1 @@
+export * from "./renderer";
```
Comment (julienblin) [inline packages/gluestack-ui/src/r5/index.ts:2]: Don't forget to also re-export anything in `./data-display` - to export the gluestackrendererprops as well.

File: packages/gluestack-ui/src/r5/data-display/fhir-value.tsx [hunk 3]
```diff
@@ -0,0 +1,22 @@
+import { FhirValueRendererProps } from "@bonfhir/react/r5";
+import { Text } from "@gluestack-ui/themed";
+import { ReactElement } from "react";
+
+export function GlueStackFhirValue(
+  props: FhirValueRendererProps<GlueStackFhirValueProps>,
+): ReactElement | null {
+  return (
+    <Text
+      className={props.className}
```
Comment (julienblin) [inline packages/gluestack-ui/src/r5/data-display/fhir-value.tsx]: Does gluestack actually support className? Only when running in web, maybe?

**Review by julienblin — CHANGES_REQUESTED** [review]
Thanks @nickvallee for your contribution.

A couple of comments:
 - I see this is a new branch - what are you going to do with the `145-create-a-bonfhirgluestack-ui-package`?
  - please run the `pnpm build` prior to committing - this is missing some files in the `r4b` folder (which tells me this package was probably not tested...)
  - please cleanup the commit history and add a changeset - [see details here](https://bonfhir.dev/docs/contributing#changesets---versioning-packages) (now that we have a documentation...)
  - and finally, link the issue (#145) to the PR

File: packages/gluestack-ui/src/r5/data-display/fhir-value.tsx (hunk 3 above)
Comment (nickvallee) [reply packages/gluestack-ui/src/r5/data-display/fhir-value.tsx]: very good point. There was no error when testing in the ecpo app, but I will remove it

**Comment by nickvallee** [comment]
@julienblin documentation was really helpful thank you. 

As for the other branch, I will remove it now 👍 . All other changes have been done, I believe. ready for second pass of review.

**Review by julienblin — APPROVED** [review]
Alright - this is good thank you!
I just pushed a small commit to add "react" as a peer dependency: https://github.com/bonfhir/bonfhir/pull/173/commits/ecb9fb0cb474609ced7b70ae43431901ef3d6f68

Without this, the react package was bundled in the final build (visible if you open the result of the build in the `dist` directory).

Please rebase and commit! The resulting package should be pushed to the npm registry automatically!

**Comment by nickvallee** [comment]
@julienblin Thanks! Glad you caught that dependency issue. Back from vacation.

I have rebased the branch and committed the changes remotely.

> Please rebase and commit! The resulting package should be pushed to the npm registry automatically!

When you tell me to rebase and commit here, do you also mean merge the PR? I imagine NPM registry will only be pushed to the npm registry once we merge this PR into the main branch?

Just wanted to double check in case you prefer to be the one who merges the PRs. Happy to merge otherwise!

**Comment by julienblin** [comment]
> @julienblin Thanks! Glad you caught that dependency issue. Back from vacation.
> 
> I have rebased the branch and committed the changes remotely.
> 
> > Please rebase and commit! The resulting package should be pushed to the npm registry automatically!
> 
> When you tell me to rebase and commit here, do you also mean merge the PR? I imagine NPM registry will only be pushed to the npm registry once we merge this PR into the main branch?
> 
> Just wanted to double check in case you prefer to be the one who merges the PRs. Happy to merge otherwise!

@nickvallee thanks - yeah go ahead and merge! Your assumption about packages and npm registry is mostly correct.
I tried to document the release process here: https://bonfhir.dev/docs/contributing#changesets---versioning-packages

**Pull request merged**